documentation/resource-definitions.md

```diff
@@ -0,0 +1,118 @@
+# Resource definitions 
+
+## V1 version
+If V1 version of Resource is needed, please add the following in the `definitions` section of swagger spec.
+
+```
+	"Resource": {
+          "properties": {
+            "id": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource Id"
+            },
+            "name": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource name"
+            },
+            "type": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource type"
+            },
+            "location": {
+              "type": "string",
+              "description": "Resource location"
+            },
+            "tags": {
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              },
+              "description": "Resource tags"
+            }
+          },
+          "required": [
+            "location"
+          ],
+          "x-ms-azure-resource": true
+        }
+```
+
+## V2 version
+If V2 version of Resource is needed, please add the following in the `definitions` section of swagger spec.
+
+```
+	"Resource": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id"
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The sku of the created namespace"
+        }
+      },
+	"Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the notification hub sku",
+          "enum": [
+            "Free",
```

Comment:
Are these SKUs always present? I realize that modelAsString is set to true, so it is probably OK to say this assuming that the vast majority of resources will follow this SKU naming pattern...